src/actions/attack.hpp

```diff
@@ -281,6 +281,8 @@ void attack_unit_and_advance(const map_location& attacker,
  */
 int under_leadership(const unit &u, const map_location& loc, const_attack_ptr weapon = nullptr, const_attack_ptr opp_weapon = nullptr);
 
+std::pair<int, std::string> under_alignment_abilities(const unit &u, const map_location& loc, int combat_modifier, bool is_fearless = false, bool use_gender = false);
```

Comment:
Documentation would help for reviewing - would drafting it in French and then getting @CelticMinstrel or myself to translate work for you?

There's another comment about whether this should just calculate the alignment, leaving it to the caller to calculate the combat bonus for that alignment.

Instead of returning the alignment as a string, please use the `unit_alignments` templated enum.

Reply:
have you see sceenshot here:: https://github.com/newfrenchy83/wesnoth/pull/79 ?

Reply:
I meant documentation for the function itself, describing what the arguments are and what it returns, just the part of the calculation that's done in this function. I know the functional currently does most of the calculation, but this fits with my other comments about trying to make this smaller by only calculating the alignment in this function, and leaving it to the caller to use another function to calculate the bonus.